### PR TITLE
databases-week3

### PR DESCRIPTION
**Pull request by Merhawi-gebreslasie**

### CODE REVIEW

File: databases/week3/hyf_lesson3.sql [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+USE hyf_lesson3;
+SET NAMES 'utf8mb4';
+CREATE TABLE `meal`(
+    `id` INT(10) UNSIGNED NOT NULL AUTO_INCREMENT,
+    `title` VARCHAR(255) NOT NULL,
+    `description` TEXT NULL DEFAULT NULL,
+    `location` VARCHAR(255) NOT NULL,
+    `when` DATETIME NOT NULL,
+    `max_reservations ` INT(10) UNSIGNED NOT NULL,
+    `price` DECIMAL NOT NULL,
```
Comment (martamatos) [inline databases/week3/hyf_lesson3.sql]: don't forget to specify how many digits you want in total and how many decimal places.

File: databases/week3/hyf_lesson3.sql [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+    `created_date` DATE NOT NULL,
+    PRIMARY KEY(`created_date`)
+) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_unicode_ci;
+CREATE TABLE `reservation`(
+    `id` INT(10) UNSIGNED NOT NULL AUTO_INCREMENT,
+    contact_name ` VARCHAR(255) NOT NULL,
```
Comment (martamatos) [inline databases/week3/hyf_lesson3.sql]: You're missing the opening ``` here.
Also, avoid the empty spaces in the column names.

File: databases/week3/hyf_lesson3.sql [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    ` contact_email ` VARCHAR(255) NOT NULL,
+    ` contact_phonenumber ` VARCHAR(255) NULL,
+    ` created_date ` DATE NOT NULL,
+     ` number_of_guests ` INT(10) UNSIGNED NOT NULL,
+    ` meal_id ` INT(10) UNSIGNED NOT NULL,
+    PRIMARY KEY(` id `),
+    CONSTRAINT ` fk_meal ` FOREIGN KEY (` meal_id `) REFERENCES ` meal `(` id `) ON DELETE CASCADE
```
Comment (martamatos) [inline databases/week3/hyf_lesson3.sql]: If we delete all reservations for a given meal, the meal entry will also be deleted, is that what you want? (It is ok, just checking ;) )
Also, what happens if you update the id of the meal?

File: databases/week3/hyf_lesson3.sql [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+    ` title ` varchar(255) NOT NULL,
+    ` description ` TEXT NULL DEFAULT NULL,
+    ` created_date ` DATE NOT NULL,
+     ` stars `INT(10) UNSIGNED NULL,
+     ` meal_id ` INT(10) UNSIGNED NOT NULL,
+    PRIMARY KEY(` id `),
+    CONSTRAINT ` fk_review_meal ` FOREIGN KEY (` meal_id `) REFERENCES ` meal `(` id `) ON DELETE CASCADE
+) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_unicode_ci;
+INSERT INTO ` meal `(
+        ` id `,
```
Comment (martamatos) [inline databases/week3/hyf_lesson3.sql]: Remember, the `id`  is auto-incremened by MySQL, so you shouldn't insert it.

File: databases/week3/hyf_lesson3.sql [hunk 5]
```diff
@@ -0,0 +1,196 @@
+CREATE DATABASE hyf_lesson3;
```
Comment (martamatos) [inline databases/week3/hyf_lesson3.sql]: When i try to run the code here i get a bunch of errors, please fix it so i can run it without problems :)

File: databases/week3/additional_meal_queries.sql [hunk 6]
```diff
@@ -0,0 +1,52 @@
+USE hyf_lesson3;
+
+SELECT *
+FROM  meal 
+WHERE  price  < 40;
+SELECT *
```
Comment (martamatos) [inline databases/week3/additional_meal_queries.sql]: Having available reservations basically means that the sum of all guests for a given meal is less than max_reservations.

This is the trap query btw. You need to sum the number_of_guests over all reservations for a given meal and then compare that number with the max_reservations of that meal.

There are at least two ways of doing this, both involve stuff that we didnt talk about in class:
 - use the COALESCE() function: https://dev.mysql.com/doc/refman/8.0/en/comparison-operators.html#function_coalesce
 - use GROUP BY, SUM, and `HAVING: https://www.w3schools.com/sql/sql_having.asp

Also remember that there might be meals that don't have any reservations yet.

File: databases/week3/additional_meal_queries.sql [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+   meal.title ,
+   meal.description ,
+   meal.location ,
+   meal.when,
+   meal.max_reservations ,
+   meal.price ,
+   meal.created_date 
+FROM  meal 
+  JOIN  review  ON  meal.id  =  review.meal_id 
+WHERE  revie.stars  > 4;
```
Comment (martamatos) [inline databases/week3/additional_meal_queries.sql:31]: typo

**Review by martamatos — CHANGES_REQUESTED** [review]
Good work overall Merhawi :)

Remember to leave empty lines between queries to make it easier to read.

And please make sure that all queries run without errors so i can test.

File: databases/week3/additional_meal_queries.sql (hunk 7 above)
Comment (Merhawi-gebreslasie) [reply databases/week3/additional_meal_queries.sql:31]: Thanks Marta ,I have made the changes.

**Pull request merged**